uploader/entities/locations.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from typing import List
+
+import pandas as pd
+from django.core.exceptions import ValidationError
+
+from location.models import CofkCollectLocation, CofkUnionLocation
+from uploader.entities.entity import CofkEntity
+from uploader.models import CofkCollectUpload
+
+log = logging.getLogger(__name__)
+
+
+class CofkLocations(CofkEntity):
+
+    def process_places_sheet(self) -> List[tuple]:
+        """
+        Get all people from people spreadsheet
+        Populating a list of tuples of (Name, iperson_id)
+        TODO where does Geonames lookup happen?
+        """
+        sheet_locations = []
+        for i in range(1, len(self.sheet_data.index)):
+            self.row_data = {k: v for k, v in self.sheet_data.iloc[i].to_dict().items() if v is not None}
+
+            self.check_data_types('Places')
+            location_id = self.row_data['location_id'] if 'location_id' in self.row_data else 1
+
+            location = CofkCollectLocation(**self.row_data)
+
+            if location_id and isinstance(location_id, int):
+                location.union_location = CofkUnionLocation.objects.filter(location_id=location_id).first()
+
+            location.location_id = location_id
+            location.element_1_eg_room = 0
+            location.element_2_eg_building = 0
+            location.element_3_eg_parish = 0
+            location.element_4_eg_city = 0
+            location.element_5_eg_county = 0
+            location.element_6_eg_country = 0
+            location.element_7_eg_empire = 0
```

Comment:
yes, it's not ideal, but I am not certain how best to provide defaults for the collection entities because it is very limited what is provided in the spreadsheet.

I suggest we keep this for now and I will imrpove